routes/api/branches.js

```diff
@@ -1,8 +1,67 @@
 const express = require("express");
 const router = express.Router();
 
-router.get("/test", (req, res) =>
-  res.json({ msg: "This is the branches route" })
-);
+const Branch = require("../../models/Branch");
+const TextString = require('../../models/TextString');
+
+// Show
+router.get("/:id", (req, res) => {
```

Comment:
This is relevant to all api modules but I'm commenting here since this is the first one:
We should separate the routes thing with the actual logic as much as we can. Why? because suppose you want to use the logic of find branch by id in another route - but you're already running in the node backend. You don't want to call the branch api with a fetch request, because there's no need - you're already running in the backend.
So what I suggest is to have the following structure:
* `routes` folder, with a file for each model
* `api` folder (parallel to `routes` folder), with a folder for each model.

Then, what you will have in the routes file, will look something like this:
```suggestion
router.get("/:id", (req, res) => {
  api.branches.get(req.params.id)
    .then((branch) => {
      if (branch) {
        res.json(branch);
      } else {
        res.status(404).json("No Such Branch");
      }
    })
    .catch((err) => res.status(404).json(err));
});
```

and in the `api/branch.js`, you will have:
```
function getBranch(id) {
    return Branch.findById(id)
}

module.exports = {
   get: getBranch
}
```
This way, two routes or two modules in your server can call the `getBranch` api and share that logic.

Reply:
do you mean the api folder will have a file for each model or a folder? I don't understand how it's supposed to work with a folder for each model